.github/workflows/release-50_publish-docker.yml

```diff
@@ -0,0 +1,204 @@
+name: Release - Publish Docker Image
+
+# This workflow listens to pubished releases or can be triggered manually.
+# It includes releases and rc candidates.
+# It fetches the binaries, checks sha256 and GPG
+# signatures, then builds an injected docker
+# image and publishes it.
+
+on:
+  release:
+    types:
+      - published
+  workflow_dispatch:
+    inputs:
+      release_id:
+        description: |
+          Release ID.
+          You can find it using the command:
+          curl -s \
+            -H "Authorization: Bearer ${GITHUB_TOKEN}" https://api.github.com/repos/$OWNER/$REPO/releases | \
+            jq '.[] | { name: .name, id: .id }'
+        required: true
+        type: string
+      image_type:
+        description: Type of the image to be published
+        required: true
+        default: rc
+        type: choice
+        options:
+          - rc
+          - release
+      registry:
+        description: Container registry
+        required: true
+        type: string
+        default: docker.io
+      owner:
+        description: Owner of the container image repo
+        required: true
+        type: string
+        default: parity
+
+env:
+  RELEASE_ID: ${{ inputs.release_id }}
+  ENGINE: docker
+  REGISTRY: ${{ inputs.registry }}
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  DOCKER_OWNER: ${{ inputs.owner || github.repository_owner }}
+  REPO: ${{ github.repository }}
+  BINARY: polkadot-parachain
+  EVENT_ACTION: ${{ github.event.action }}
+  EVENT_NAME: ${{ github.event_name }}
+  IMAGE_TYPE: ${{ inputs.image_type }}
+
+jobs:
+  fetch-artifacts:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout sources
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+
+      - name: Prepare temp folder
+        run: |
+          TMP=$(mktemp -d)
+          echo "TMP=$TMP" >> "$GITHUB_ENV"
+          pwd
+          ls -al "$TMP"
+
+      - name: Fetch lib.sh from polkadot repo
+        working-directory: ${{ env.TMP }}
+        run: |
+          curl -O -L \
+              -H "Accept: application/vnd.github.v3.raw" \
+              https://raw.githubusercontent.com/paritytech/polkadot/master/scripts/ci/common/lib.sh
+
+          chmod a+x lib.sh
+          ls -al
+
+      - name: Fetch release artifacts based on final release tag
+        if: ${{ env.EVENT_NAME == 'release' && env.EVENT_ACTION != '' && env.EVENT_ACTION == 'published' }}
+        run: |
+          mkdir -p release-artifacts && cd release-artifacts
+
+            for f in $BINARY $BINARY.asc $BINARY.sha256; do
+                URL="https://github.com/${{ github.event.repository.full_name }}/releases/download/${{ github.event.release.tag_name }}/$f"
+                echo " - Fetching $f from $URL"
+                wget "$URL" -O "$f"
+            done
+            chmod a+x $BINARY
+            cp -f ${TMP}/lib.sh .
+            ls -al
```

Comment:
Some indentation does not look right here :)

Reply:
good catch, it's addressed :)